README.md

```diff
@@ -10,4 +10,449 @@ TODO: description
+
+**Returns**
+
+- A success message
+
+**Throws**
+
+- `401` if the user is not logged in
+- `403` if the user is not the owner of the medication
+- `404` if the medicationId is invalid
+- `409` if the medicationId is already deactivated 
+
+### Entry
+
+#### `GET /api/entries` - Get all entries of the user
+
+**Returns**
+- The array of entries by the user sorted in descending order by date
+
+**Throws** 
+- `401` if the user is not logged in
+
+#### `POST /api/entries` - Make a new entry in the log
+
+**Body**
+- `type` _{string}_ - The type of the log entry
+- `detail` _{string}_ - The detail of the log entry
+- `notes` _{string}_ - The log entry's notes (optional)
+- `date` _{Date}_ - The date associated with the log entry (optional)
+
+**Returns**
+- A success message
+- The newly created entry object
+
+**Throws** 
+- `401` if the user is not logged in
+- `400` if the `type` or `detail` is empty or a stream of empty spaces
+
+#### `PATCH /api/entries/:entryId` - Editing an entry in the log
+
+**Body**
+- `type` _{string}_ - The type of the log entry (optional)
+- `detail` _{string}_ - The detail of the log entry (optional)
+- `notes` _{string}_ - The log entry's notes (optional)
+- `date` _{Date}_ - The date associated with the log entry (optional)
+
+**Returns**
+- A success message
+- The updated entry object
+
+**Throws** 
+- `401` if the user is not logged in
+- `404` if the `entryId` is invalid
+- `403` if the user is not the owner of the entry
+
+#### `DELETE /api/entries/:entryId` - Delete an entry in the log
+
+**Returns**
+- A success message
+
+**Throws** 
+- `401` if the user is not logged in
+- `404` if the `entryId` is invalid
+- `403` if the user is not the owner of the entry
+
+
+### Trend
+
+#### `GET /api/trends` - Get all trends for a given user  
+**Returns**  
+- An array of trends created by user with given username (taken from session)  
+**Throws**  
+- `401` if `user` is not found in session  
+
+#### `GET /api/trends/:itemId` - Get set of points that reference a given item (data entry)  
+**Returns**  
+- An array of points  
+**Throws**  
+- `401` if the user is not logged in  
+- `403` if the user is not the author of the point  
+- `404` if `item` is invalid  
+
+#### `GET /api/trends` - Get trend (set of points) that includes a given set of items (log entries)  
+**Body**  
+- `items` _{array<Entry>}_ - Set of items to be plotted in the trend  
+- `title` _{string}_ - The title of the graph  
+- `start_date` _{date}_ - Date from which to start plotting the trend graph  
+- `end_date` _{date}_ - Date at which to finish plotting the trend graph  
+**Returns**  
+- An array of points (one trend)  
+**Throws**  
+- `400` if the items, title, start_date, or end_date is empty/a stream of empty spaces  
+- `401` if the user is not logged in  
+
+#### `POST /api/trends` - Add a point to a trend  
+**Body**  
+- `item` _{Entry}_ - Log entry to be added to trend  
```

Comment:
Entry isn't a json type so doesn't make sense to have in the api spec. I would just call it a string which would correspond to the id of the entry.